ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java

```diff
@@ -260,6 +275,18 @@ public byte[] getBytesForHash() {
         return allBytes;
       }
     }
+
+    public byte[] getBytesForIdentity() {
+      if (inputSplit instanceof HashableInputSplit) {
+        return ((HashableInputSplit) inputSplit).getBytesForHash();
+      } else {
+        byte[] pathBytes = getPath().toString().getBytes();
```

Comment:
not sure if that's an optimal approach. for example iceberg uses reverse order comparison: https://github.com/apache/iceberg/blob/a8ec43d975d8d3bbacb6880b487208b00cb7361f/api/src/main/java/org/apache/iceberg/types/Comparators.java#L364

Reply:
serialization overhead shouldn't be noticeable, right?

Reply:
@okumin, don't we have the same functionality already, HiveInputFormat#getBytesForHash()